nectar/downloaders/threaded.py

```diff
@@ -182,16 +187,14 @@ def _download_one(self, request):
         :return:    download report
         :rtype:     nectar.report.DownloadReport
         """
-        session = build_session(self.config)
-        return self._fetch(request, session)
+        build_session(self.config, self.session)
```

Comment:
Yeah, that is more clear. I really didn't like how I implemented this, but without re-writing nectar...